src/Compilers/CSharp/Test/Emit/Emit/EmitMetadataTests.cs

```diff
@@ -2936,5 +2936,329 @@ public class Child : Parent, IParent
+                			IL_0005: ldc.i4.5
+                			IL_0006: call string '<PrivateImplementationDetails>'::BytesToString(uint8*, int32)
+                			IL_000b: stsfld string '<PrivateImplementationDetails>'/'<S>1BFD09D1A433FB78117B4C7B1583D16D'::s
+                			IL_0010: ret
+                		} // end of method '<S>1BFD09D1A433FB78117B4C7B1583D16D'::.cctor
+                	} // end of class <S>1BFD09D1A433FB78117B4C7B1583D16D
+                	// Fields
+                	.field assembly static initonly valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=5' '185F8DB32271FE25F561A6FC938B2E264306EC304EDA518007D1764826381969' at I_{{offset}}
+                    .data cil I_{{offset}} = bytearray (
+                		48 65 6c 6c 6f
+                	)
+                	// Methods
+                	.method private hidebysig static 
+                		string BytesToString (
+                			uint8* bytes,
+                			int32 length
+                		) cil managed 
+                	{
+                		// Method begins at RVA 0x207b
+                		// Code size 13 (0xd)
+                		.maxstack 8
+                		IL_0000: call class [mscorlib]System.Text.Encoding [mscorlib]System.Text.Encoding::get_UTF8()
+                		IL_0005: ldarg.0
+                		IL_0006: ldarg.1
+                		IL_0007: callvirt instance string [mscorlib]System.Text.Encoding::GetString(uint8*, int32)
+                		IL_000c: ret
+                	} // end of method '<PrivateImplementationDetails>'::BytesToString
+                } // end of class <PrivateImplementationDetails>
+                """);
+        }
+
+        [Theory, CombinatorialData]
+        public void DataSectionStringLiterals_MetadataOnly(
+            [CombinatorialValues("0", "off")] string feature)
+        {
+            var source = """
+                System.Console.WriteLine("Hello");
+                """;
+            CompileAndVerify(source,
+                emitOptions: EmitOptions.Default.WithEmitMetadataOnly(true),
+                parseOptions: TestOptions.Regular.WithFeature("experimental-data-section-string-literals", feature),
+                verify: Verification.FailsPEVerify with
+                {
+                    PEVerifyMessage = """
+                        Bad token as entry point in CLR header.
+                        """,
+                },
+                symbolValidator: static (ModuleSymbol module) =>
+                {
+                    AssertEx.AssertEqualToleratingWhitespaceDifferences("""
+                        <Module>
+                        EmbeddedAttribute
+                        RefSafetyRulesAttribute
+                        Program
+                        """, module.TypeNames.Join("\n"));
+                })
+                .VerifyDiagnostics();
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

I think we need to test the following aspects:
- sharing of an "RVA" field type for strings of the same length but different in value
- sharing of an "RVA" field type with an array initializer of the same length but different value
- sharing of an "RVA" field by different literals of equal value
- sharing of an "RVA" field with an array initializer
- sharing of an "<S>..." type by different literals of equal value